capstone/src/main/java/com/google/step/servlets/RestaurantServlet.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.GeoPt;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.step.data.Restaurant;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for sending/getting restaurants to/from Datastore. */
```

Comment:
I guess my thinking for what I implemented was to have doGet retrieve all the restaurants we have available (we could put a limit on this as well, I just didn't for the prototype) and return them as a list of Restaurants to the frontend. Then, they are all displayed on the "landing page" when you first open the page, and they all have an onClick action that uses the Restaurant object from the list to populate the details page for that restaurant. That way, there's restaurants to look at when you visit the site even before you start searching, and you don't have to query for the restaurant whenever you click on one wanting to see more details. 
If we instead wanted doGet to retrieve a single restaurant, how would we be able to populate the initial page before searching? would that be a different servlet?

Reply:
Is the search page different from the landing page? I'd imagine it's the same thing (feed of restaurants), so depending on what's happening with the search indexing, it might make sense to have the landing page be a basic search query with nothing but location populated. Basically, I'm trying to keep it so there's only every one implementation of every type of thing ("list of restaurants" being one such category).